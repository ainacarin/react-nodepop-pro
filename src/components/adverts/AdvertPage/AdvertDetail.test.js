import React from 'react';
import { shallow } from 'enzyme';

import AdvertDetail from './AdvertDetail';
import { ConfirmationButton } from '../../shared'

describe('AdvertDetail', () => {
  const props = {
    name: 'advert detail',
    sale: true,
    price: 45.25,
    tags: ['tag1','tag2'],
    photo: null,
    onDelete: jest.fn()
  };

  const render = () => shallow(<AdvertDetail {...props} />);

  test('should render', () => {
    const wrapper = render();
    expect(wrapper.exists()).toBe(true);
  });

  test('snapshot testing', () => {
    const wrapper = render();
    expect(wrapper).toMatchSnapshot();
  });

  test('snapshot find element ConfirmationButton', () => {
    const expectedConfirmationText = 'Are you sure?';
    const wrapper = render();
    const confirmationButton = wrapper.find('ConfirmationButton');
    const { confirmation } = confirmationButton.props();
    expect(confirmation).toEqual(expectedConfirmationText);
  });

});
