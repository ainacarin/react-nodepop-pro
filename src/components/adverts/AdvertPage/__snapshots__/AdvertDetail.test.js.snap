// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdvertDetail snapshot testing 1`] = `
<div>
  <p>
    advert detail
  </p>
  <p>
    Sell
  </p>
  <p>
    tag1, tag2
  </p>
  <p>
    45.25
  </p>
  <img
    alt="advert detail"
    height="200"
    src="placeholder.png"
    style={
      Object {
        "objectFit": "contain",
      }
    }
    width="200"
  />
  <ConfirmationButton
    confirmation="Are you sure?"
    onConfirm={[MockFunction]}
  >
    Delete
  </ConfirmationButton>
</div>
`;
